src/Microsoft.ML.TimeSeries/AdaptiveSingularSpectrumSequenceModeler.cs

```diff
@@ -22,35 +29,29 @@ namespace Microsoft.ML.Transforms.TimeSeries
     /// This class implements basic Singular Spectrum Analysis (SSA) model for modeling univariate time-series.
     /// For the details of the model, refer to http://arxiv.org/pdf/1206.6910.pdf.
     /// </summary>
-    internal sealed class AdaptiveSingularSpectrumSequenceModeler : SequenceModelerBase<Single, Single>
+    public sealed class AdaptiveSingularSpectrumSequenceModeler : ICanForecast<float>
     {
-        internal const string LoaderSignature = "SSAModel";
-
+        /// <summary>
+        /// Ranking selection method for the signal.
+        /// </summary>
         public enum RankSelectionMethod
         {
             Fixed,
             Exact,
             Fast
         }
 
-        internal sealed class SsaForecastResult : ForecastResultBase<Single>
-        {
-            public VBuffer<Single> ForecastStandardDeviation;
-            public VBuffer<Single> UpperBound;
-            public VBuffer<Single> LowerBound;
-            public Single ConfidenceLevel;
-
-            internal bool CanComputeForecastIntervals;
-            internal Single BoundOffset;
-
-            public bool IsVarianceValid { get { return CanComputeForecastIntervals; } }
-        }
-
+        /// <summary>
+        /// Growth ratio. Defined as Growth^(1/TimeSpan).
+        /// </summary>
         public struct GrowthRatio
         {
             private int _timeSpan;
             private Double _growth;
 
+            /// <summary>
+            /// Time span of growth ratio. Must be strictly positive.
+            /// </summary>
             public int TimeSpan
```

Comment:
Why use an `int` here? Why not use a `TimeSpan` type, which avoids any confusion about what unit this `int` is? seconds? Minutes? days?